app/src/main/java/com/graeberj/taskman/auth/domain/repository/AuthRepositoryImpl.kt

```diff
@@ -14,7 +15,13 @@ class AuthRepositoryImpl(private val api: ApiService) : AuthRepository {
             val result = api.loginUser(loginRequestDto)
             Resource.Success(data = result.toLoggedInUser())
         } catch (e: Exception) {
-            Resource.Error(message = e.toString())
+            if (e == CancellationException()){
+                throw e
+            } else {
+                Resource.Error(message = e.toString())
+                //once I'm more comfortable with error handling, I'll implement some sort of custom
+                // error handling
+            }
```

Comment:
This code seems very repetitive. I personally created such a helper function for that:
```
suspend inline fun <T> authenticatedRetrofitCall(
    jsonSerializer: JsonSerializer,
    crossinline doCall: suspend () -> AuthResult<T>
): AuthResult<T> {
    return try {
        doCall()
    } catch(e: HttpException) {
        e.printStackTrace()
        e.asAuthResult(jsonSerializer)
    } catch(e: IOException) {
        e.printStackTrace()
        AuthResult.Error(UiText.unknownError())
    }
}

suspend fun HttpException.getErrorMessage(serializer: JsonSerializer): ErrorMessage? {
    return withContext(Dispatchers.IO) {
        response()?.errorBody()?.bytes()?.decodeToString()?.let { body ->
            serializer.fromJson(body, ErrorMessage::class.java)
        }
    }
}

suspend fun <T> HttpException.asAuthResult(serializer: JsonSerializer): AuthResult<T> {
    if(isUnauthorized()) {
        return AuthResult.Unauthorized()
    }
    val message = getErrorMessage(serializer)?.message
    return AuthResult.Error(
        message = if(message != null) {
            UiText.DynamicString(message)
        } else UiText.unknownError()
    )
}
```
Can then be called like this:
```
val result = authenticatedRetrofitCall(serializer) {
    api.getEvent(eventId)
    AuthResult.Success(Unit)
}
```